mw/i2c_service/controller/pca9685/controller.cpp

```diff
@@ -109,21 +106,68 @@ std::vector<uint8_t> PCA9685::GenerateData(const uint8_t &channel, const uint16_
     PRESCALE_REG, 121,  // przeskalowanie dla 50 Hz
     static_cast<uint8_t>(LED0_ON_L+4*channel), 0x0 & 0xFF,  // ON LOW REG Val
     static_cast<uint8_t>(LED0_ON_H+4*channel), 0x0 >> 8,   // ON HIGH REG Val
-    static_cast<uint8_t>(LED0_OFF_L+4*channel), static_cast<uint8_t>(pos & 0xFF),  // OFF LOW REG Val
+    static_cast<uint8_t>(LED0_OFF_L+4*channel), static_cast<uint8_t>(pos & 0x00FF),  // OFF LOW REG Val
     static_cast<uint8_t>(LED0_OFF_H+4*channel), static_cast<uint8_t>(pos >> 8)};   // OFF HIGH REG Val;
 }
 
 core::ErrorCode PCA9685::SetServo(uint8_t channel, uint16_t pos) {
   return this->i2c_->Write(PCA9685_ADDRESS, GenerateData(channel, pos));
 }
+uint8_t PCA9685::CalculateFirstRegister(const uint8_t& channel) {
+  return 6 + 4 * channel;
+}
+uint8_t PCA9685::CalculatePosition(const std::vector<uint8_t>& val) {
+    const uint8_t lsb =  val[2];
+    const uint8_t msb = val[3];
```

Comment:
czy jest mozliwość że rozmiar bedzie mniejszy ??

Reply:
![image](https://github.com/user-attachments/assets/3320c550-e188-49d7-a5c8-c806f720ba1c)
jak sie rozmiar nie będzie zgadzał to zwraca optional. wiec tu raczej ten sprawdzenie rozmiaru jest zbędne